### ISSUE TITLE
Compare-IntuneBackupFile does not compare sub properties

### ISSUE THREAD

**Issue by ErikvNoorden** [jseerden/IntuneBackupAndRestore#5]
The Compare-IntuneBackupFile function only loops through the toplevel properties.

eg: 
```
...
"applicationGuardForceAuditing":  false,
"applicationGuardBlockClipboardSharing":  "notConfigured",
"applicationGuardAllowPrintToPDF":  false,
"bitLockerSystemDrivePolicy":  {
    "encryptionMethod":  "xtsAes128",
    "prebootRecoveryUrl":  null,
    "recoveryOptions":  {
        "blockDataRecoveryAgent":  false,
         "recoveryPasswordUsage":  "allowed",
         "recoveryKeyUsage":  "allowed",
         "hideRecoveryOptions":  true,
         "enableRecoveryInformationSaveToStore":  true,
         "recoveryInformationToStore":  "passwordAndKey",
         "enableBitLockerAfterRecoveryInformationToStore":  true
     }
},
...
```
in given example, everything under bitLockerSystemDrivePolicy is ignored for comparison.
would be nice if it could loop through nested properties as well

**Comment by jseerden** [comment]
Thanks for opening this issue. Will look into this.

**Comment by jseerden** [comment]
This issue is fixed in the v1.3.0 release
